app/services/legacy/dataset_import_service.rb

```diff
@@ -103,7 +104,7 @@ def base_attributes_for_resource(resource)
       format: resource["format"],
       name: datafile_name(resource),
       created_at: resource["created"] || dataset.created_at,
-      updated_at: dataset.last_updated_at
+      updated_at: resource["last_modified_at"] || resource["created"]
```

Comment:
Should this be `resource["last_modified_at"] || resource["created"] || dataset.created_at`?

There looks to be a case where `resource["created"]` can be nil given the way `created_at` is set.